lstchain/reco/dl1_to_dl2.py

```diff
@@ -402,11 +403,33 @@ def build_models(filegammas, fileprotons,
 
     reg_energy = train_energy(df_gamma_reg, custom_config=config)
 
+    if save_models:
+        os.makedirs(path_models, exist_ok=True)
+
+        file_reg_energy = path_models + "/reg_energy.sav"
+        joblib.dump(reg_energy, file_reg_energy, compress=3)
+    if free_model_memory:
```

Comment:
There is not really a reason to put this behind a flag, is there? The variable is not used after dumping.

Reply:
For some reason the function returns the models (probably useful if not saved). I personally don't know if there are any use case. If not we could remove save_models and free_model_memory

Reply:
It is at least used in test currently